File: Problems/CCO/cco07p2.java
import java.util.*;
import java.io.*;

public class cco07p2 {
	public static void main(String[] args) {
		FastReader sc = new FastReader();
		Set<String> set = new HashSet<String>();
		int n = sc.nextInt();
		for(int i=1; i<=n; i++) {
			int arr[] = new int[6];
			for(int j=0; j<6; j++) {
				arr[j] = sc.nextInt();
			}
			Arrays.sort(arr);
			String s = "";
			for(int j=0; j<6; j++) {
				s+=arr[j];
			}
			set.add(s);
			if(set.size()!=i) {
				System.out.println("Twin snowflakes found.");
				System.exit(0);
			}
		}
		
		System.out.println("No two snowflakes are alike.");
	}

	static class FastReader {
		BufferedReader br;
		StringTokenizer st;

		public FastReader() {
			br = new BufferedReader(new InputStreamReader(System.in));
		}

		String next() {
			while (st == null || !st.hasMoreElements()) {
				try {
					st = new StringTokenizer(br.readLine());
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
			return st.nextToken();
		}

		int nextInt() {
			return Integer.parseInt(next());
		}

		long nextLong() {
			return Long.parseLong(next());
		}

		double nextDouble() {
			return Double.parseDouble(next());
		}

		String nextLine() {
			String str = "";
			try {
				str = br.readLine();
			} catch (IOException e) {
				e.printStackTrace();
			}
			return str;
		}
	}
}

File: Problems/CCO/cco07p2.cpp
#include <iostream>
#include <set>
#include <bits/stdc++.h>
using namespace std;

int main()
{
    set<string> set1;
    
    int n;
    cin >> n;
    for(int i=0; i<n; i++){
       vector<string> arr;
        for(int j=0; j<6; j++){
            string a;
            cin >> a;
            arr.push_back(a);
            
        }
        
        vector<string> s;
         
        for(int j=0; j<2; j++){
            for(int k=0; k<6; k++){
                string str = "";
                for(int l=0; l<6; l++){
                    if (j == 1) {
						str += arr[5 - (l + k) % 6];
					} else {
						str += arr[(l + k) % 6];
					}
                }
                
                s.push_back(str);
            }
        }
        
        sort(s.begin(), s.end());
        
        string str = "";
        for(int j=0; j<12; j++){
            str+=s[j];
        }
        
        if(set1.find(str)!=set1.end()){
            cout << "Twin snowflakes found.\n";
            return 0;
        }
        
        set1.insert(str);
        
    }
    
    cout << "No two snowflakes are alike.\n";
    
    return 0;
}